src/librustc_trans/back/write.rs

```diff
@@ -154,13 +154,13 @@ fn get_llvm_opt_size(optimize: config::OptLevel) -> llvm::CodeGenOptSize {
     }
 }
 
-pub fn create_target_machine(sess: &Session) -> TargetMachineRef {
-    target_machine_factory(sess)().unwrap_or_else(|err| {
+pub fn create_target_machine(sess: &Session, find_features: bool) -> TargetMachineRef {
```

Comment:
What's the  `find_features` parameter about?

Reply:
`create_target_machine` calls `is_pie_binary` which requires `crate_types`. However when we are discovering the target CPU features, `crate_types` is uninitialized. So we pass in `find_features` to let `create_target_machine` assume that we aren't looking at a pie binary.

Reply:
Ah, OK. Could you add a comment explaining that?

Reply:
Done.